src/shapes/box2d_shape.h

```diff
@@ -15,14 +15,19 @@ class Box2DCollisionObject;
 
 class Box2DShape {
 	RID self;
+	Box2DCollisionObject *body;
```

Comment:
I do not understand this. The `Box2DShape` is supposed to be a resource (hence the `RID`) that can be shared by multiple bodies. For example it should be possible to create 10 bodies all referring to one and the same circle shape resource. Can you test an example like that? The rest of the PR looks very good, but I don't think this part can work like this.

Reply:
Hm, that would be a problem indeed. I guess it would have a reference to only one of the bodies.
Right now this is used to keep a track of the body, and it's used in case of eg. `_intersect_shape` to return collider_id and collider.
However, I see now that there is a method on the b2Fixture called GetBody, so I wouldn't need this reference, as I could use that.
Good catch.

Reply:
@Ughuuu indeed, that sounds like the correct approach.